### DISCUSSION TITLE
study-masters-or-no/

### DISCUSSION THREAD

**Discussion by giscus** [amir-pourmand/hugo-papermod-farsi-template#23] [Announcements]
# study-masters-or-no/

راستش اخیراً چند نفر ازم این سؤال رو پرسیدند و دیدم جوابم داره یکسان میشه. گفتم شاید بد نباشه، بشینم اینجا راجع بهش صحبت کنم.
البته که معتقدم جواب من، ارزش چندانی نداره؛ چون نه مطالعه خاصی در این مورد داشتم. نه کار آماری انجام دادم. نه در این زمینه تخصص دارم.
ولی بالاخره الان شما تو کوچه هم رد بشید و یه نفر رو رندوم انتخاب کنید، یک سری حرف‌ها راجع به این سوال داره.

https://aprd.ir/study-masters-or-no/

**Comment by saeedrafieyan** [comment]
سلام
وقتت بخیر، من از طریق سایت محمدرضا باهات آشنا شدم و این پست رو هم اتفاقی دیدم. نکته‌ی جالبش اینه که من هم مثل تو اعزام یک آبان بنیادم. خوشحال میشم اونجا همدیگه رو پیدا کنیم و گپ بزنیم :)))

**Reply by pourmand1376** [reply]
سلام. 
چه دنیای کوچیکی داریم. حتماً :)

**Comment by MortezaAfroozeh** [comment]
سلام:)
وسط تصمیم هاتون برای ارشد خوندن هیچ وقت به خوندن اقتصاد یا مدیریت (یا کلا هر رشته دیگه ای ...} فکر کردین؟
اگه اره چیشد منصرف شدین؟

**Reply by pourmand1376** [reply]
سلام مرتضی. 

راستش نه‌. زمانی که از کار قبلیم بیرون زدم، فقط به هوش مصنوعی فکر می‌کردم و گزینه‌هایی مثل مدیریت یا اقتصاد رو در نظر نگرفتم‌.

هر چند الان هم به مدیریت خیلی علاقه دارم و دارم از طریق متمم دوره می‌گذرونم.

**Reply by MortezaAfroozeh** [reply]
اخه شدیدا بین چند راهی رشته های مختلف (مدیریت و اقتصاد و هوش و بیوانفورماتیک) واسه ارشد گیرم.
البته امید دارم تا موقع ارشدم که چند سالی مونده این فضای پر ابهام تا حد زیادی واسم رفع بشه.
به هر حال خیلی خیلی نوشته هات کمکم میکنه و خیلییییییی ازت ممنون اقا امیر.

**Comment by sealimousavi** [comment]
اگر کسی کارشناسی مهندسی کامپیوتر نخونده باشه و دوست داشته باشه به دلایل یک و دو جمع بندی بیاد ارشد کامپیوتر بخونه نظرت چیه؟ به نظرت ممکنه سخت باشه براش با وجود اینکه کار کرده باشه و علاقه هم داشته باشه؟

**Reply by pourmand1376** [reply]
سلام علی. 

دو تا نکته راجع به مطلبی که گفتی به ذهنم می‌رسه:

یکی این که معمولاً بچه‌هایی که کارشناسی دانشگاه‌های خوب و رشته اپلای‌خور رفتند (به درستی) انگیزه‌ای برای ارشد خوندن تو ایران ندارند. اینها می‌خوان اپلای کنند برای کارشناسی ارشد و خیلی‌هاشون هم دکترا مستقیم میرن می‌خونن. پس اینها برای کنکور کامپیوتر تلاشی نمی‌کنند و رقیب تو حساب نمیشن. 

من تو بچه‌های کامپیوتر شریف به وضوح می‌دیدم که تعداد بسیار کمی به کنکور ارشد فکر می‌کردند. فضای کارشناسی اونجا اینطوریه که هر کی اپلای نکنه و نره، انگار یک چیزی کم داشته یا داره. وقتی وارد دانشکده کامیپوتر میشی، رایج‌ترین سؤالی که یکی ممکنه ازت بپرسه، اینه چه چرا هنوز از این خراب‌شده نرفتی؟

پس یک دسته‌ بزرگ از افرادی که پرتلاش هستند و بازی کنکور رو بلدند (تو کارشناسی هم رتبه خوب آوردند)، دیگه وارد این بازی نمی‌شن. 

پس می‌مونن افرادی که یا از رشته‌های دیگه میان و افرادی که از دانشگاه‌های دیگه میان که رقابت کردن با اونها تو کنکور نسبتاً راحت‌تره. انگار همه تقریباً از یک جا شروع می‌کنند. 

بگذار شرایط خودم رو مثال بزنم که دقیق متوجه بشی چی می‌گم. زمان ما ۱۰ تا درس رو باید برای کامپیوتر می‌خوندیم ([+](https://aprd.ir/computer-engineering-masters-exam/)). یکیش که انگلیسی بود. مدار منطقی و معماری کامپیوتر و سیستم عامل رو که تقریباً سر کلاس هیچی یاد نگرفته بودم. اینه که کاملاً از ابتدا شروع کردم. 

راجع به نظریه و کامپایلر و هوش‌ مصنوعی و داده الگوریتم هم یادمه که فضای کلاس ما با فضای چیزی که تو تست ازمون خواسته میشد بسیار متفاوت بود. پس عملاً مزیتی از این نظر نداشتم. شما حتی اگر الان کورس هوش مصنوعی دکتر رهبان (تو کارشناسی) رو نگاه کنید که به نظرم کورس خیلی خوبیه، لزوماً نمی‌تونید بهتر تست بزنید؛ چون چیزهایی که درس میدن (حتی از نظر سیلابس) متفاوته. حالا اصلاً این بماند که بعد ۲ سال اکثر چیزهایی که یاد میدن رو آدم فراموش می‌کنه. 

خلاصه این که که کلاً دو تا درس شبکه و پایگاه داده بود که یه کم سطحم بالاتر بود و زودتر می‌تونستم جمع‌بندی کنم. بقیه درس‌ها مثل تو و خیلی از بچه‌های دیگه بودم.

نکته دوم هم این که دانشگاه‌های ایران مثل قیف برعکسه. ورود بهشون ممکنه سخت باشه ولی فارغ‌التحصیلی ازشون اصلاً سخت نیست. حالا یکی زودتر فارغ میشه یکی دیرتر؛ ولی من نفهمیدم تا حالا که کسی که تلاش خیلی خیلی عادی‌ای انجام میده، مشکلی داشته باشه. فقط مسئله ورودش هست وگرنه وقتی وارد شدی، می‌بینی که همه مثل من و تو هستند.

**Comment by Mohammad0713** [comment]
سلام آقا امیر خوشحالم که باهات از طریق سایت آشنا شدم یسری سوال ازات داشتم واقعا هفته ها ذهنمو درگیر کرده یکی اینکه من یکم اعتماد به نفسم پایینه هر وقت به دانشگاه های تاپ فکر میکنم مثلا میگم منو چه به دانشگاه هایی مثل تهران یا شریف یا امیرکبیر که برچسب نخبه و المپیادی روشون خورده و از نظر سطح علمی بالاهستند با خودم فکر میکنم بر فرض هم اگه تلاش کنم و دانشگاه تاپ قبول بشم آیا میتونم با این افراد رقابت کنم و کم نیارم؟حالا اینکه چرا میخام دانشگاه تاپ قبول بشم به دو دلیله یکی جایگاه اجتماعی و دوم ساخت رزومه قوی از طرفی میترسم بعداز فارغ التحصیلی در مقطع ارشد در تئوری بمونم و نتونم کار درستی پیدا کنم و این دوراهی اصلی منه چون میدونیم کار یا بهتر بگم بدست آوردن مهارت های بازار کار همزمان با ارشد خوندن در دانشگاه تاپ خیلی خیلی سخته نظر شما چیه ؟شما بین قبولی در دانشگاه تاپ یا قبولی در دانشگاه خوب ولی ضعیف تر مثل یزد(الان یادم اومد همشهری هستیم) ودر کنارش یاد گرفتن مهارت های بازار کار کدوم رو انتخاب میکنید البته اینم بگم که من قصد اپلای یا خوندن دکتری رو ندارم به نظر شما در شرایط بنده بهترین تصمیم چی میتونه باشه ؟سوال دومم اینه در مقطع ارشد تا چه حد زبان نیازمون میشه؟ من زبانم متوسطه بنظرتون برای اینکه در مقطع ارشد بابت زبان به مشکل برنخورم چطور برنامه ریزی کنم ؟ درآخر هم ببخشید که خیلی طولانی شد چون واقعا ذهنمو درگیر کرده بود امیدوارم وقت ارزشمندتان رو بزارید و جواب بدید چون مطمئنم حرف های شاید بتونه سرنوشت من رو تغییر بده

**Reply by pourmand1376** [reply]
سلام محمد جان. 

راجع به حرف‌هات چند تا نکته به ذهنم می‌رسه:

فقط قبلش بگم که تجربیات من اکثراً محدود به رشته مهندسی کامپیوتر هست و شاید تو رشته‌های دیگه چندان کاربرد نداشته باشه. 

نکته اول-  منم قبل از این که شریف بیام فکر می‌کردم که شریف خبریه! فکر می‌کردم که تو دانشگاه‌های برتر ایران، بچه‌ها نشستند و دارند روی مرزهای علم کار می‌کنند. انقدر هم همشون خفن هستند که از دانشگاه‌های خارج میان و این نخبه‌ها رو می‌قاپند و بورسیه می‌کنند و ازشون خواهش می‌کنند که برن آمریکا و کانادا دکترا بخونند. فکر می‌کردم نمی‌تونم باهاشون رقابت کنم و عقب می‌مونم. 

طولی نکشید تا وارد شریف شدم و فهمیدم این خبرا هم نیست. بچه‌هایی که اونجا بودند از کره مریخ نبودند. همین مثل من و تو بودند. هوش‌شون هم عجیب غریب بالا نبود. ولی یک تفاوت بارز داشتند. اکثراً خیلی تلاش می‌کردند. 

همین بود که انقدر دانشگاه‌های مختلف اپلای می‌کردند تا یه جا بتونند پذیرش بگیرند. اگر می‌خواستند اپلای کنند به هر دری می‌زدند و از هر کسی پول قرض می‌گرفتند که بتونند برن. آخرش هم واقعاً می‌رفتند چون واقعاً می‌خواستند که برن. 

نکته دوم- دانشگاه‌های ایران (چه تو پزشکی و چه تو غیرپزشکی) مثل قیف برعکس می‌مونه. این اصطلاح رو از یکی شنیدم ولی یادم نمیاد کی بود. منظور از قیف برعکس اینه که ورود بهش دریچه نسبتاً تنگی داره ولی خروج ازش اصلاً سخت نیست. 

از بین همکلاسی‌های من، همه (بدون حتی یک استثناء) فارغ‌التحصیل شدند. خیلی خیلی باید بی‌خیال درس و دانشگاه باشی که نتونی با درس‌ها کنار بیای. بالاخره ممکنه یکی دو نمره بالاتر پایین‌تر بگیری؛ ولی خیالی نیست. تهش این نمره‌ها و معدل‌ها (خصوصاً برای تو که نمی‌خوای اپلای کنی)، هیچ تأثیری نداره. 

خلاصه این که به محض این که وارد شدی، اگر خودت رو عمداً به بی‌خیالی نزنی یا این که برچسب‌های سیاسی نخوری، با احتمال خیلی زیاد می‌تونی مثل بقیه درست رو تموم کنی و بری دنبال کارت. 

نکته سوم- واقعیتش اگر دانشجوی کارشناسی بودی، به نظرم حتی دانشگاهی مثل دانشگاه یزد سطح خوبی داره؛ چون اکثر بچه‌هایی که میان سطح خوبی دارند و تلاش‌گر هستند. اما ارشد احتمالاً انقدر خوب نیست. 

در واقع اکثر (نمیگم همه) کسانی که کارشناسی تهران می‌خونند، چندان نیازی به ارشد خوندن ندارند و اگر هم بخونند، اکثراً میرن خارج کشور. 

پس بچه‌های دانشگاه‌های متوسط‌تر فرصت این رو پیدا می‌کنند که به دانشگاه‌های برتر محلق بشن. همین‌طور الی اخر. 

نکته چهارم- بخش بزرگی از صنعت ایران هم داخل شهر تهران هست. به همین خاطر موقعیت‌های شغلی اونجا بیشتره و دانشگاه‌های تهران ارتباطات خیلی خوبی برات میارن. خیلی بهتر از اون چیزی که می‌تونی تصورش رو بکنی. اتفاقاً خیلی‌ها هم دنبال کار هستند و در حین تحصیل کار هم می‌کنند. اساتید هم مشکلی با کار کردن‌شون ندارند. خود اساتید هم اکثراً شرکت دارند و دعوتت می‌کنند که تو شرکت‌شون کار کنی. 

همچنین دانشگاه‌های خوب (مثل شریف و امیرکبیر و تهران و حتی علم و صنعت) برچسب خیلی خوبی برات هستند که تو اوایل دوران کاری کمک‌کننده هستند. هر چند بعد از مدتی این برچسب‌ها هم رنگ می‌بازند و خود واقعی آدم نمایان میشه. 

نکته پنجم- احتمالاً حالت به هم می‌خوره اگر بهت بگم که یادگیری زبان انگلیسی از اوجب واجبات هست. از بس که تو سرمون تکرار کردند. واقعاً هم هست. منم نمی‌خوام تکرار کنم. چه ارشد و چه تو بازار کار به شدت نیازه. حداقل رشته ما این‌طور بود که اکثر کتاب‌های مرجع انگلیسی بودند و ترجمه هم نداشتند. ولی نمی‌خواد ازش بترسی. تو دوره ارشد تقریباً همه در حال تقویت زبان هستند و کاملاً عادیه که زبانت قوی نباشه. 

نکته ششم- راجع به برنامه‌ریزی و هدف‌گذاری هم چی از این بهتر که به معلمم، محمدرضا شعبانعلی ارجاع بدم. دوره هدف‌گذاری‌شون رو خودم هم دارم گوش میدم و تو وبلاگ‌شون کلی راجع به برنامه‌ریزی و مسائل مشابه صحبت کردند.

**Reply by Mohammad0713** [reply]
ممنون از وقتی که گذاشتید و جواب دادید نکات خیلی مفیدی فرمودین که من چند دفعه خوندم از اونجایی که منم کامپیوتر میخونم و اتفاقا به هوش مصنوعی هم علاقمندم میخاستم ازتون بپرسم کلا گرایش هوش در مقطع ارشد چجوریه ؟و اگه توصیه هایی در این زمینه دارید بسیار مشتاق شنیدن اون هستم و اینکه به نظرتون درحال حاضر کدوم یکی از فیلد های هوش مصنوعی بازار کار روبه رشد و بهتری در ایران داره؟

**Reply by pourmand1376** [reply]
جواب سؤالت طولانیه :)) اجازه بده سرفرصت یک پست باید در این زمینه بنویسم.

**Reply by Mohammad0713** [reply]
آها چشم پس مشتاقانه منتظر میمونم بازم تشکر میکنم ازتون که این تجربیات گران بها رو در اختیار من و دیگران قرار میدید تا با دید بهتر کاری رو انجام بدیم
